File: duck/duck1.java
class Duck {
    String name;
    String color;

    public void quack() {
        System.out.println("quack by duck");
    }
}

class GreeHeadDuck extends Duck {
    public void quack() {
        System.out.println("quack by green head duck");
    }
}

public class Example {
    public static void main(String[] args) {
        Duck duck = new Duck();
        duck.quack();

        Duck greeheadduck = new GreeHeadDuck();
        greeheadduck.quack();
    }
}

File: duck/duck1.py
class Duck:
    def __init__(self, name):
        self.name = name

    def quack(self):
        print("c c c")

    def fly(self):
        print("i can fly")


class GreenHeadDuck(Duck):
    def __init__(self, name):
        super().__init__(name)
        self.color = "green"


class WoodDuck(Duck):
    def quack(self):
        print("i can't quack")

    def fly(self):
        print("i can't fly")


class YellowPlasticDuck(Duck):
    def quack(self):
        print("ji ji ji")

    def fly(self):
        print("i can't fly")